src/affexpr.jl

```diff
@@ -181,13 +181,15 @@ function addconstraint(m::Model, c::LinearConstraint)
     end
     return LinConstrRef(m,length(m.linconstr))
 end
-addconstraint(m::Model, c::Array{LinearConstraint}) =
+addconstraint(m::Model, c::AbstractArray{LinearConstraint}) =
     error("The operators <=, >=, and == can only be used to specify scalar constraints. If you are trying to add a vectorized constraint, use the element-wise dot comparison operators (.<=, .>=, or .==) instead")
 
-function addVectorizedConstraint(m::Model, v::Array{LinearConstraint})
-    ret = Array{LinConstrRef}(size(v))
-    for I in eachindex(v)
-        ret[I] = addconstraint(m, v[I])
+function addVectorizedConstraint(m::Model, v::AbstractArray{LinearConstraint})
```

Comment:
This doesn't seem like the right fix. I think ``constructconstraint!(x::Array, sense::Symbol) = map(c->constructconstraint!(c,sense), x)`` should be changed to return a flat ``Vector`` of constraints. Then there's no need to touch ``addVectorizedConstraint``.